### PR TITLE
Upgrade James SHA-1

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
08:05:23.482 [ERROR] o.a.j.GuiceJamesServer - Fatal error while starting James
com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT10S
	at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.lambda$scheduleTimeout$1(CqlRequestHandler.java:207)
	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.run(HashedWheelTimer.java:715)
	at io.netty.util.concurrent.ImmediateExecutor.execute(ImmediateExecutor.java:34)
	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.expire(HashedWheelTimer.java:703)
	at io.netty.util.HashedWheelTimer$HashedWheelBucket.expireTimeouts(HashedWheelTimer.java:790)
	at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:503)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```
Seeing the log before this being 
```
08:05:04.885 [WARN ] o.a.j.CONFIGURATION - Could not find extra system properties file conf/jvm.properties
```
My guess is on that :)

**Comment by Arsnael** [comment]
I need this to finalize this upgrade: https://github.com/apache/james-project/pull/1203

**Review by chibenwa — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Comment by Arsnael** [comment]
```
[ERROR] Errors: 
[ERROR]   DistributedImapAndSmtpTest>DeploymentValidation.validateDeploymentWithMailsFromSmtp:98 » ConditionTimeout
```
I got that twice in a row so I guess it's an issue... note that other tests are passing... will investigate -_-

**Comment by Arsnael** [comment]
So this test `DistributedImapAndSmtpTest>DeploymentValidation.validateDeploymentWithMailsFromSmtp` is indeed failing, can reproduce locally.

When debugging checking the James logs, after the smtp message sent and being succesfully added to the queue in rabbitmq, you can see this error message in the logs:

```
09:28:58.526 [ERROR] c.r.c.i.ForgivingExceptionHandler - Consumer com.rabbitmq.client.impl.recovery.AutorecoveringChannel$2@2c18fa65 (amq.ctag-vE4Xtupcj9HHsbS8qFVkXg) method handleDel
ivery for channel AMQChannel(amqp://guest@192.168.160.2:5672/,14) threw an exception for channel AMQChannel(amqp://guest@192.168.160.2:5672/,14)                                       
java.lang.reflect.InaccessibleObjectException: Unable to make field private final byte[] java.lang.String.value accessible: module java.base does not "opens java.lang" to unnamed module @4c3b6dbe                                                                                                                                                                           
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)                                                                                          
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)                                                                                          
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Unknown Source)                                                                                                     
        at java.base/java.lang.reflect.Field.setAccessible(Unknown Source)                                                                                                             
        at org.nustaq.serialization.FSTClazzInfo.createFieldInfo(FSTClazzInfo.java:512)                                                                                                
        at org.nustaq.serialization.FSTClazzInfo.createFields(FSTClazzInfo.java:368)     
        at org.nustaq.serialization.FSTClazzInfo.<init>(FSTClazzInfo.java:129)        
        at org.nustaq.serialization.FSTClazzInfoRegistry.getCLInfo(FSTClazzInfoRegistry.java:129)
        at org.nustaq.serialization.FSTClazzNameRegistry.addClassMapping(FSTClazzNameRegistry.java:98)                                                                                 
        at org.nustaq.serialization.FSTClazzNameRegistry.registerClassNoLookup(FSTClazzNameRegistry.java:85)       
        at org.nustaq.serialization.FSTClazzNameRegistry.registerClass(FSTClazzNameRegistry.java:81)
        at org.nustaq.serialization.FSTConfiguration.addDefaultClazzes(FSTConfiguration.java:807)
        at org.nustaq.serialization.FSTConfiguration.initDefaultFstConfigurationInternal(FSTConfiguration.java:477)
        at org.nustaq.serialization.FSTConfiguration.createDefaultConfiguration(FSTConfiguration.java:472)
        at org.nustaq.serialization.FSTConfiguration.createMinBinConfiguration(FSTConfiguration.java:229)
        at org.nustaq.serialization.FSTConfiguration.constructJsonConf(FSTConfiguration.java:319)
        at org.nustaq.serialization.FSTConfiguration.createJsonConfiguration(FSTConfiguration.java:301)
        at org.nustaq.serialization.FSTConfiguration.createJsonConfiguration(FSTConfiguration.java:290)
        at org.nustaq.serialization.FSTConfiguration.createJsonConfiguration(FSTConfiguration.java:294)
        at org.apache.mailet.Serializer$FSTSerializer.<clinit>(Serializer.java:578)
        ... 39 common frames omitted
Wrapped by: java.lang.ExceptionInInitializerError: null
        at org.apache.mailet.Serializer$Registry.<clinit>(Serializer.java:79)
        at org.apache.mailet.AttributeValue.findSerializerAndDeserialize(AttributeValue.java:206)
        at org.apache.mailet.AttributeValue.lambda$deserialize$1(AttributeValue.java:202)
        at java.base/java.util.Optional.flatMap(Unknown Source)
        at org.apache.mailet.AttributeValue.lambda$deserialize$2(AttributeValue.java:202)
        at java.base/java.util.Optional.flatMap(Unknown Source)
        at org.apache.mailet.AttributeValue.deserialize(AttributeValue.java:201)
        at java.base/java.util.Optional.flatMap(Unknown Source)
        at org.apache.mailet.AttributeValue.fromJson(AttributeValue.java:191)
        at org.apache.mailet.AttributeValue.fromJsonString(AttributeValue.java:175)
        at org.apache.james.queue.rabbitmq.MailReferenceDTO.lambda$mailMetadata$4(MailReferenceDTO.java:228)
        at com.github.fge.lambdas.consumers.BiConsumerChainer.lambda$sneakyThrow$14(BiConsumerChainer.java:73)
        at com.google.common.collect.RegularImmutableMap.forEach(RegularImmutableMap.java:292)
        at org.apache.james.queue.rabbitmq.MailReferenceDTO.mailMetadata(MailReferenceDTO.java:231)
        at org.apache.james.queue.rabbitmq.MailReferenceDTO.toMailReference(MailReferenceDTO.java:207)
        at org.apache.james.queue.rabbitmq.MailLoader.lambda$load$0(MailLoader.java:50)
        at reactor.core.publisher.MonoCallable.call(MonoCallable.java:92)
        at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:127)
        at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
        at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
        at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:129)
        at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816)
        at reactor.core.publisher.MonoCallable.subscribe(MonoCallable.java:62)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4400)
        at reactor.core.publisher.FluxMergeSequential$MergeSequentialMain.onNext(FluxMergeSequential.java:237)
        at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.onNext(FluxFilterFuseable.java:118)
        at reactor.core.publisher.FluxUsing$UsingConditionalSubscriber.onNext(FluxUsing.java:510)
        at reactor.core.publisher.FluxCreate$BufferAsyncSink.drain(FluxCreate.java:814)
        at reactor.core.publisher.FluxCreate$BufferAsyncSink.next(FluxCreate.java:739)
        at reactor.core.publisher.FluxCreate$SerializedFluxSink.next(FluxCreate.java:161)
        at reactor.rabbitmq.Receiver.lambda$null$9(Receiver.java:216)
        at com.rabbitmq.client.impl.recovery.AutorecoveringChannel$2.handleDelivery(AutorecoveringChannel.java:588)
        at com.rabbitmq.client.impl.ConsumerDispatcher$5.run(ConsumerDispatcher.java:149)
        at com.rabbitmq.client.impl.ConsumerWorkService$WorkPoolRunnable.run(ConsumerWorkService.java:104)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
```
When looking at rabbitmq queues in the admin panel, indeed you can see there is a mail dangling in the queue there and not being consumed. But well, normal that consumer crashed...
 
So... I tried many things, also of course thought of the schenanigans java 11 - java 17 explaining why the last commit... (as I get the error message `java.lang.reflect.InaccessibleObjectException: Unable to make field private final byte[] java.lang.String.value accessible: module java.base does not "opens java.lang" to unnamed module @4c3b6dbe `) but even with that locally... I still get that same error message.

Honestly I'm at a dead end... if someone has an idea, I'm listening...

**Review by chibenwa — APPROVED** [review]

File: tmail-backend/apps/distributed-es6-backport/pom.xml [hunk 1]
```diff
@@ -414,7 +414,21 @@
                             <port>8000</port> <!-- WEBADMIN -->
                         </ports>
                         <appRoot>/root</appRoot>
+                        <jvmFlags>
+                            <jvmFlag>-Dlogback.configurationFile=/root/conf/logback.xml</jvmFlag>
+                            <jvmFlag>-Dworking.directory=/root/</jvmFlag>
+                            <!-- Prevents Logjam (CVE-2015-4000) -->
+                            <jvmFlag>-Djdk.tls.ephemeralDHKeySize=2048</jvmFlag>
+                            <jvmFlag>-Dextra.props=/root/conf/jvm.properties</jvmFlag>
```
Comment (quantranhong1999) [inline tmail-backend/apps/distributed-es6-backport/pom.xml:422]: Can you try adding this:
```
                                <jvmFlag>--add-opens=java.base/java.lang=ALL-UNNAMED</jvmFlag>
                                <jvmFlag>--add-opens=java.base/java.math=ALL-UNNAMED</jvmFlag>
                                <jvmFlag>--add-opens=java.base/java.util=ALL-UNNAMED</jvmFlag>
                                <jvmFlag>--add-opens=java.base/java.util.concurrent=ALL-UNNAMED</jvmFlag>
                                <jvmFlag>--add-opens=java.base/java.net=ALL-UNNAMED</jvmFlag>
                                <jvmFlag>--add-opens=java.base/java.text=ALL-UNNAMED</jvmFlag>
```

File: tmail-backend/apps/distributed/pom.xml [hunk 2]
```diff
@@ -409,7 +429,21 @@
                             <port>8000</port> <!-- WEBADMIN -->
                         </ports>
                         <appRoot>/root</appRoot>
+                        <jvmFlags>
+                            <jvmFlag>-Dlogback.configurationFile=/root/conf/logback.xml</jvmFlag>
+                            <jvmFlag>-Dworking.directory=/root/</jvmFlag>
+                            <!-- Prevents Logjam (CVE-2015-4000) -->
+                            <jvmFlag>-Djdk.tls.ephemeralDHKeySize=2048</jvmFlag>
+                            <jvmFlag>-Dextra.props=/root/conf/jvm.properties</jvmFlag>
```
Comment (quantranhong1999) [inline tmail-backend/apps/distributed/pom.xml:437]: And here also. I am kinda sure `java.lang.reflect.InaccessibleObjectException` related to Java 17.

File: tmail-backend/apps/distributed-es6-backport/pom.xml (hunk 1 above)
Comment (Arsnael) [reply tmail-backend/apps/distributed-es6-backport/pom.xml]: yes good point... actually it's that commit that is the problem. I don't know why you need the flags declared here for that too but it's a good remark. But by just kicking this out it works as well. So I reopened https://github.com/linagora/tmail-backend/pull/469 where I will apply your suggestion instead, to speed things up. 

But thanks for taking a look :)

Comment (quantranhong1999) [reply tmail-backend/apps/distributed-es6-backport/pom.xml]: >  I don't know why you need the flags declared here for that too but it's a good remark

I think somehow now rabbitmq client production code call some reflection code that is not enabled by default on Java 17. So actually our product' JRE needs to support that reflection, not just in the maven test JRE.

**Review by chibenwa — COMMENTED** [review]
Haaaaaa...

**Comment by chibenwa** [comment]
\o/ kodoo to you @Arsnael and the team!

**Pull request merged**